### PR TITLE
docs(root): add testing documentation for Jest

### PR DESCRIPTION
**Pull request by GCHQ-Developer-112**
## Summary of the changes
Add testing documentation for Jest. Update nav priority for addition of new page

## Related issue
#96

## Checklist
- [x] I have manually accessibility tested any changes, if relevant.

### CODE REVIEW

File: src/content/structured/get-started/testing-components.mdx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+### Step four
+
+Finally, run your tests using `npm test` or `yarn test`.
+
+More information on setting up Jest can be found on [Jest’s Getting Started page](https://jestjs.io/docs/getting-started).
+
+## What is the shadow DOM?
+
+A key part of our packages is the shadow DOM. With the shadow DOM, we can encapsulate each component’s markup structure, styling and behaviour so different parts don’t clash.
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:55]:
```suggestion
A key part of our components is the shadow DOM. With the shadow DOM, we can encapsulate each component’s markup structure, styling and behaviour so different parts don’t clash.
```

File: src/content/structured/get-started/testing-components.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+According to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), the “shadow DOM allows hidden DOM trees to be attached to elements in the regular DOM tree — this shadow DOM tree starts with a shadow root, underneath which you can attach any element, in the same way as the normal DOM.”
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:57]:
```suggestion
According to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), the “shadow DOM allows hidden DOM trees to be attached to elements in the regular DOM tree — this shadow DOM tree starts with a shadow root, which you can attach any element, in the same way as the normal DOM.”
```

File: src/content/structured/get-started/testing-components.mdx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Testing shadow components
+
+In the DOM, elements can be found using `document.querySelector`. Shadow components can be found in a similar way using the `shadowRoot` prefix. The shadow root is the root node of the shadow DOM tree.
+
+Note: components that having been slotted in can be found in DOM. They are not a part of their parent component's shadow DOM
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:63]:
```suggestion
Note: components that have been slotted in can be found in the 'light' DOM. They are not a part of their parent component's shadow DOM
```

File: src/content/structured/get-started/testing-components.mdx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+### Example
+
+```js
+// To find all navigation items in a top nav
+document.querySelector(“ic-top-navigation).querySelectorAll(“ic-navigation-item”)
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:69]:
```suggestion
document.querySelector("ic-top-navigation").querySelectorAll(“ic-navigation-item”)
```

File: src/content/structured/get-started/testing-components.mdx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+// Find the search button for the search bar in the top nav (nested shadow DOM)
+document.querySelector("ic-top-navigation > ic-search-bar").shadowRoot.querySelector("ic-button#search-submit-button").shadowRoot.querySelector("#ic-button-with-tooltip-search-submit-button")
+```
+
+## Example test
+
+Below is an example of a React component that uses the `IcTopNavigation` component. The labels for the navigation items are being mapped to the `IcNavigationItem` component
+
+```js
+<---topNav.js--->
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:83]:
```suggestion
// topNav.js
```

File: src/content/structured/get-started/testing-components.mdx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+import React from 'react';
+import { IcTopNavigation, IcNavigationItem, IcSearchBar } from "@ukic/react";
+
+export const topNav = () => {
+    const navItemLabels = ["About", "Drinks", "Recipes", "Blog"];
+
+    const navItems = navItemsLabels.map((navItemLabel, index) =>
+        {return (
+            <IcNavigationItem key={index} label={navItemLabel} href={`/${navItemLabel}`} slot="navigation">
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:92]:
```
 const navItems = navItemsLabels.map((navItemLabel, index) => (
            <IcNavigationItem key={index} label={navItemLabel} href={`/${navItemLabel}`} slot="navigation">
```

File: src/content/structured/get-started/testing-components.mdx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    )
+}
+```
+
+We are now going to test that the labels are being mapped correctly using the Arrange-Act-Assert pattern.
+
+Jest has a range of [matchers](https://jestjs.io/docs/expect) to choose from for your test assertions. Below, we make use of the `.toMatch` matcher for our test assertion.
+
+```js
+<---topNav.test.js--->
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:112]:
```suggestion
// topNav.test.js
```

File: src/content/structured/get-started/testing-components.mdx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    const navItems = Array.from(
+          document.querySelectorAll("ic-top-navigation ic-navigation-item")
+        ).map((i) =>
+          i.shadowRoot.querySelector(".ic-tooltip").textContent
+        );
+
+    // Assert
+    navItems.forEach((navItem, i) => {
+        expect(navItem).toMatch(navItemsLabels[i]);
+      });
```
Comment (jd239) [inline src/content/structured/get-started/testing-components.mdx:128]: Indentation is quite right

File: src/content/structured/get-started/testing-components.mdx (hunk 3 above)
Comment (jd239) [reply src/content/structured/get-started/testing-components.mdx:63]: Had a quick look for a website we could refer to for a definition of the 'light' DOM'. Perhaps reference this: https://web.dev/shadowdom-v1/#terminology-light-dom-vs-shadow-dom

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Review by jd239 — CHANGES_REQUESTED** [review]
A couple comments

File: src/content/structured/get-started/testing-components.mdx (hunk 6 above)
Comment (GCHQ-Developer-112) [reply src/content/structured/get-started/testing-components.mdx:92]: Why would this use regular brackets if it's an arrow function? Using the brackets clears the red error brackets I was getting but I'm not sure why this is

File: src/content/structured/get-started/testing-components.mdx (hunk 4 above)
Comment (GCHQ-Developer-847) [inline src/content/structured/get-started/testing-components.mdx:69]:
```suggestion
document.querySelector(“ic-top-navigation).querySelectorAll("ic-navigation-item")
```

**Review by jd239 — APPROVED** [review]

File: src/content/structured/get-started/testing-components.mdx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+In the DOM, elements can be found using `document.querySelector`. Shadow components can be found in a similar way using the `shadowRoot` prefix. The shadow root is the root node of the shadow DOM tree.
+
+Note: components that have been slotted in can be found in the 'light' DOM. They are not a part of their parent component's shadow DOM.
+
+### Example
+
+```js
+// To find all navigation items in a top nav
+document.querySelector(“ic-top-navigation).querySelectorAll("ic-navigation-item")
```
Comment (ad9242) [inline src/content/structured/get-started/testing-components.mdx:71]:
```suggestion
document.querySelector("ic-top-navigation").querySelectorAll("ic-navigation-item")
```

Comment (GCHQ-Developer-112) [reply src/content/structured/get-started/testing-components.mdx]: I remember fixing this issue. Don't know what happen 😢

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**